Dockerfile

```diff
@@ -134,49 +149,47 @@ RUN conda install -y python=3.7.1 \
 RUN python -c "from matplotlib import font_manager" && \
     sed -i 's/\(backend *: \).*$/\1Agg/g' $( python -c "import matplotlib; print(matplotlib.matplotlib_fname())" )
 
-# Installing Ubuntu packages and cleaning up
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends \
-                    git=1:2.7.4-0ubuntu1 && \
-    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-
-# Install latest pandoc
-RUN curl -o pandoc-2.2.2.1-1-amd64.deb -sSL "https://github.com/jgm/pandoc/releases/download/2.2.2.1/pandoc-2.2.2.1-1-amd64.deb" && \
-    dpkg -i pandoc-2.2.2.1-1-amd64.deb && \
-    rm pandoc-2.2.2.1-1-amd64.deb
-
 # Unless otherwise specified each process should only use one thread - nipype
 # will handle parallelization
 ENV MKL_NUM_THREADS=1 \
     OMP_NUM_THREADS=1
 
-WORKDIR /root/
-
 # Precaching atlases
-ENV CRN_SHARED_DATA /templateflow
-ADD docker/scripts/get_templates.sh get_templates.sh
-RUN mkdir $CRN_SHARED_DATA && \
-    /root/get_templates.sh && \
-    find $CRN_SHARED_DATA -type d -exec chmod 555 {} \; && \
-    find $CRN_SHARED_DATA -type f -exec chmod 444 {} \; && \
-    chmod +w $CRN_SHARED_DATA
+WORKDIR /opt
+ENV TEMPLATEFLOW_HOME="/opt/templateflow"
+RUN pip install "datalad==0.10.0" && \
+    rm -rf ~/.cache/pip
+
+RUN git config --global user.name "First Last" && \
+    git config --global user.email "email@domain.com" && \
+    datalad install -r https://github.com/templateflow/templateflow.git
+RUN datalad get $TEMPLATEFLOW_HOME/tpl-MNI152NLin2009cAsym/*_T1w.nii.gz \
+                $TEMPLATEFLOW_HOME/tpl-MNI152NLin2009cAsym/*_desc-brain_mask.nii.gz \
+                $TEMPLATEFLOW_HOME/tpl-MNI152Lin/*_T1w.nii.gz \
+                $TEMPLATEFLOW_HOME/tpl-MNI152Lin/*_desc-brain_mask.nii.gz \
+                $TEMPLATEFLOW_HOME/tpl-OASIS30ANTs/*_T1w.nii.gz \
+                $TEMPLATEFLOW_HOME/tpl-OASIS30ANTs/tpl-OASIS30ANTs_res-01_desc-brain_mask.nii.gz \
+                $TEMPLATEFLOW_HOME/tpl-OASIS30ANTs/tpl-OASIS30ANTs_res-01_label-brain_probseg.nii.gz \
+                $TEMPLATEFLOW_HOME/tpl-OASIS30ANTs/tpl-OASIS30ANTs_res-01_desc-BrainCerebellumExtraction_mask.nii.gz
```

Comment:
Just to make sure, is it still the case that if we *didn't* pre-fetch these, s/fMRIPrep would still download them as needed?

Reply:
Yes.

Although there are two known issues:
- Race conditions - when running in parallel (I think this happened in niworkflows) datalad chokes. This does not refer to datalad's capabilities of parallel downloads, this is when you run datalad.get on the same file twice. This was one reason to pre-cache everything we probably need. But running, e.g., with the skull-stripping with NKI as reference (so not cached), would work (download, and then keep cached).
- This pre-fetching forced me to stop running the containers as user. People who want to run them as user will need to define a `$TEMPLATEFLOW_HOME` variable. I haven't actually tried this, and would also require revising permissions on `/templateflow` to allow reuse by the user.

But yeah, the central functionality is working well (or better said, datalad works well)